### PR TITLE
Move account in memory caches to accounts DB

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary

This PR will remove `sequenceToNoteHashes` and `nonChainNoteHashes`. These two are being moved to database stores that will be fetched from using range queries.

## Testing Plan

 - Added new tests
 - Rerun migration

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[x] Yes
```

### CODE REVIEW

File: ironfish/src/wallet/database/accountsdb.ts [hunk 1]
```diff
@@ -65,6 +69,16 @@ export class AccountsDB {
     value: Buffer
   }>
 
+  sequenceToNoteHash: IDatabaseStore<{
+    key: [Account['prefix'], [number, TransactionHash]]
```
Comment (hughy) [inline ironfish/src/wallet/database/accountsdb.ts:73]: `TransactionHash` should be some kind of note hash type or Buffer

File: ironfish/src/wallet/database/accountsdb.ts [hunk 2]
```diff
@@ -123,6 +137,22 @@ export class AccountsDB {
...
+      keyEncoding: new PrefixEncoding(
+        new PrefixEncoding(new BufferEncoding(), U32_ENCODING, 32),
+        new BufferEncoding(),
+        4,
+      ),
+      valueEncoding: NULL_ENCODING,
+    })
+
+    this.nonChainNoteHashes = this.database.addStore({
+      name: 's',
```
Comment (hughy) [inline ironfish/src/wallet/database/accountsdb.ts:151]: Duplicate name from above

**Comment by hughy** [comment]
Do we need to update the migration for these new stores?

File: ironfish/src/wallet/database/accountsdb.ts (hunk 2 above)
Comment (NullSoldier) [reply ironfish/src/wallet/database/accountsdb.ts]: fixed

File: ironfish/src/wallet/database/accountsdb.ts (hunk 1 above)
Comment (NullSoldier) [reply ironfish/src/wallet/database/accountsdb.ts]: fixed

**Pull request merged**